runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java

```diff
@@ -113,10 +118,18 @@ public final class Engine implements AutoCloseable
             tasks = newFixedThreadPool(config.taskParallelism(), this::newTaskThread);
         }
 
-        int workerCount = config.workers();
-
-        Info info = new Info(config.directory(), workerCount);
-        info.reset();
+        int workerCount;
+        if (readOnly)
+        {
+            Info info = new Info(config.directory());
+            workerCount = info.workers();
+        }
+        else
+        {
+            workerCount = config.workers();
+            Info info = new Info(config.directory(), workerCount);
+            info.reset();
+        }
```

Comment:
Suggest pushing the readonly handling inside `Info`, much like other layout builders, instead of introducing the complexity here at the call site.

Reply:
done